resources/assets/lib/beatmapsets-show/scoreboard-table-row.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import FlagCountry from 'components/flag-country';
+import Mod from 'components/mod';
+import { PlayDetailMenu } from 'components/play-detail-menu';
+import ScoreboardTime from 'components/scoreboard-time';
+import BeatmapJson from 'interfaces/beatmap-json';
+import ScoreJson from 'interfaces/score-json';
+import { route } from 'laroute';
+import { computed, makeObservable } from 'mobx';
+import { observer } from 'mobx-react';
+import core from 'osu-core-singleton';
+import * as React from 'react';
+import PpValue from 'scores/pp-value';
+import { classWithModifiers, Modifiers } from 'utils/css';
+import { hasMenu, modeAttributesMap } from 'utils/score-helper';
+
+const bn = 'beatmap-scoreboard-table';
+
+interface TdLinkProps {
+  children?: React.ReactNode;
+  href: string;
+  modifiers?: Modifiers;
+}
+
+function TdLink(linkProps: TdLinkProps) {
+  return (
+    <td className={`${bn}__cell`}>
+      <a
+        className={classWithModifiers(`${bn}__cell-content`, linkProps.modifiers)}
+        href={linkProps.href}
```

Comment:
This is always going to be the score url; I think I originally planned to use the function for other urls, but the extra stuff in those cells made it too complicated to share them 🤔

Reply:
There's no good way to share it short of using context or being part of the parent component (which means the convenient children syntax can't be used).

I don't want to do the previous "its own component but shares props with parent" thing. Creating new component class every time also seem equally shady.

Reply:
oh, it's a component 🤔